### ISSUE TITLE
Should Bitcoin Core quit when something goes wrong?

### ISSUE THREAD

**Issue by hebasto** [bitcoin/bitcoin#22398]
(Moved from https://github.com/bitcoin-core/gui/issues/364)

Bitcoin Core quits without any error message being shown. The application just disappears from the taskbar. I realize that things can go wrong (and I would very much figure out what goes wrong for me specifically), but should the application quit rather than gracefully show an error message and prompt the user to do something (for example check the error log or restart the software)? That is, is this type of behavior a bug or a feature? I definitely think that it shouldn't work this way.

**Expected behavior**

Bitcoin Core should still be open, possibly with some error message showing that it has stopped running with some suggestions what the user can do.

**Actual behavior**

Program quits and disappears from the taskbar.

**To reproduce**

I guess this depends on your system, but for me, I simply import my old wallet into the wallet folder, start Bitcoin Core and then wait. When searching for this error online, I see that it has been an issue more or less from the start.

**System information**

I run version 0.21.1, 64-bit, of Bitcoin Core, downloaded as an installer from Bitcoin.org. I'm using Windows version 10.0.19042.

https://github.com/bitcoin-core/gui/issues/364#issuecomment-861465896:
> I did try to run bitcoind instead. At first, it worked, but after trying to load my wallet via the cli, it just shut down (again, non gracefully; it just disappeared), and the cli gave me an error message saying that the daemon needed to run. Ever since that happening, running bitcoind just results in it closing itself after a while. The last thing I manage to see in the window, which I don't think should close itself, is "Rescan started from block something, something....". No extra log messages are written to either debug.log or db.log.

**Comment by SkierScottC** [comment]
I successfully installed Bitcoin Core from Bitcoin.org as well but closed it while it was synching.  Once rebooted I was able to start Bitcoin Core but it never appeared.  Pulling up the Task Manager I can see Core running but after about 30 seconds it simply disappears.  I tried uninstalling and reinstalling then removing the file directory and reinstalling but the application still will not run.  I'm assuming what is described here is the same issue I'm seeing so rather than creating a new issue I thought I'd try commenting.

My windows specs...
Edition	Windows 10 Pro
Version	21H1
OS build	19043.1083
Experience	Windows Feature Experience Pack 120.2212.3530.0

**Comment by JeremyRubin** [comment]
IMO as soon as something unexpected happens it's best to shut down as immediately as possible so that any and all processes stop occurring. We wouldn't want some runaway thread to do something bad, terminating stops all processing and clears memory.

An error message can be showed later on (e.g. next start).

**Comment by SkierScottC** [comment]
Makes sense to "log" the issue and display it upon restart.  The problem I have is that I can't use the application.  It simply will not stay running so maybe I need to create a separate issue?
